testing/integration/otel_test.go

```diff
@@ -0,0 +1,288 @@
+		apmVersionMismatch, // apm version mismatch
+		apmReadyLog,        // apm ready
+	)
+
+	// prepare agent
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// prepare input
+	agentWorkDir := fixture.WorkDir()
+	fileName := "content.log"
+	fixture.WriteFileToWorkDir(ctx, "", fileName)
+
+	testId := info.Namespace
+
+	apmConfig := fmt.Sprintf(apmOtelConfig, filepath.Join(agentWorkDir, fileName), testId)
+
+	err = fixture.Configure(ctx, []byte(apmConfig))
+	require.NoError(t, err)
+
+	componentsDir, err := aTesting.FindComponentsDir(agentWorkDir)
+	require.NoError(t, err)
+
+	// start apm default config just configure ES output
+	esHost, err := getESHost()
+	require.NoError(t, err, "failed to get ES host")
+	require.True(t, len(esHost) > 0)
+
+	esClient := info.ESClient
+	esApiKey, err := createESApiKey(esClient)
+	require.NoError(t, err, "failed to get api key")
+	require.True(t, len(esApiKey) > 1, "api key is invalid %q", esApiKey)
+
+	apmArgs := []string{
+		"run",
+		"-e",
+		"-E", "output.elasticsearch.hosts=['" + esHost + "']",
+		"-E", "output.elasticsearch.api_key=" + esApiKey,
+		"-E", "apm-server.host=127.0.0.1:8200",
+		"-E", "apm-server.ssl.enabled=false",
+	}
+
+	apmPath := filepath.Join(componentsDir, "apm-server")
+	var apmFixtureWg sync.WaitGroup
+	apmFixtureWg.Add(1)
+	apmContext, apmCancel := context.WithCancel(ctx)
+	defer apmCancel()
+	go func() {
+		aTesting.RunProcess(t,
+			logWatcher,
+			apmContext, 0,
+			true, true,
+			apmPath, apmArgs...)
+		apmFixtureWg.Done()
+	}()
+
+	// start agent
+	var fixtureWg sync.WaitGroup
+	fixtureWg.Add(1)
+	go func() {
+		fixture.RunWithClient(ctx, false)
+		fixtureWg.Done()
+	}()
+
+	err = logWatcher.WaitForKeys(context.Background(),
+		10*time.Minute,
+		500*time.Millisecond,
+		apmReadyLog,
+	)
+	require.NoError(t, err, "APM not initialized")
+	fixture.WriteFileToWorkDir(ctx, apmProcessingContent, fileName)
+
+	// check index
+	var hits int
+	match := map[string]interface{}{
+		"labels.host_test-id": testId,
+	}
+
+	// apm mismatch or proper docs in ES
+	err = logWatcher.WaitForKeys(context.Background(),
+		20*time.Second,
+		500*time.Millisecond,
+		apmVersionMismatch)
+	if err == nil {
+		// err nil means there's a match within time limit
+		// mark skipped to make it explicit it was not successfully evaluated
+		t.Skip("agent version needs to be equal to stack version")
+	} else {
+		// failed to get APM version mismatch in time
+		// processing should be running
+		require.Eventually(t,
+			func() bool {
+				docs := findESDocs(t, func() (estools.Documents, error) {
+					return estools.GetLogsForIndexWithContext(context.Background(), esClient, "logs-apm*", match)
+				})
+				hits = len(docs.Hits.Hits)
+				return hits > 0
```

Comment:
Similar to the test above, you should be verifying that the index contains what you expect it to contain, not just that it contains anything at all.